core/src/smartcontracts/isi/world.rs

```diff
@@ -368,6 +368,35 @@ pub mod isi {
             })
         }
     }
+
+    impl Execute for SetParameter<Parameter, Account> {
+        type Error = Error;
+
+        #[metrics(+"set_parameter")]
+        fn execute(
+            self,
+            _authority: <Account as Identifiable>::Id,
+            wsv: &WorldStateView,
+        ) -> Result<(), Self::Error> {
+            let new_param = self.parameter.clone();
+
+            wsv.modify_world(|world| {
+                let old_param = world.parameters.iter().find_map(|old_param| {
+                    if old_param.id() == new_param.id() {
+                        Some(world.parameters.remove(&old_param)).flatten()
+                    } else {
+                        None
+                    }
+                });
```

Comment:
Since `Parameter` is hashed and compared by it's `id` field you can directly call `world.parameters.remove(&new_param)`.